backend/de.metas.deliveryplanning.base/src/main/java/de/metas/deliveryplanning/process/M_Delivery_Planning_GenerateShortageOverage.java

```diff
@@ -0,0 +1,264 @@
+	private final ISysConfigBL sysConfigBL = Services.get(ISysConfigBL.class);
+
+	@Param(parameterName = PARAM_QTY, mandatory = true)
+	private BigDecimal p_Qty;
+
+	@Override
+	public ProcessPreconditionsResolution checkPreconditionsApplicable(final @NonNull IProcessPreconditionsContext context)
+	{
+		if (!context.isSingleSelection())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection().toInternal();
+		}
+
+		final DeliveryPlanningId deliveryPlanningId = DeliveryPlanningId.ofRepoId(context.getSingleSelectedRecordId());
+		final Optional<DeliveryPlanningReceiptInfo> optionalDeliveryPlanningReceipt = deliveryPlanningService.getReceiptInfoIfIncomingType(deliveryPlanningId);
+		if (!optionalDeliveryPlanningReceipt.isPresent())
+		{
+			return ProcessPreconditionsResolution.rejectWithInternalReason("Delivery planning is not of Type Incoming");
+		}
+
+		final DeliveryPlanningReceiptInfo receiptInfo = optionalDeliveryPlanningReceipt.get();
+		if (!receiptInfo.isReceived())
+		{
+			return ProcessPreconditionsResolution.rejectWithInternalReason("Not received");
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@Nullable
+	@Override
+	public Object getParameterDefaultValue(final IProcessDefaultParameter parameter)
+	{
+		if(PARAM_QTY.contentEquals(parameter.getColumnName()))
+		{
+			return PARAM_DEFAULT_VALUE_ZERO;
+		}
+		return IProcessDefaultParametersProvider.DEFAULT_VALUE_NOTAVAILABLE;
+	}
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		final DeliveryPlanningId deliveryPlanningId = DeliveryPlanningId.ofRepoId(getRecord_ID());
+
+		final DocTypeQueryBuilder docTypeQuery = DocTypeQuery.builder()
+			.docBaseType(MaterialPhysicalInventory)
+			.adClientId(getClientID().getRepoId())
+			.adOrgId(OrgId.ANY.getRepoId());
+
+		final AdMessageKey adMessageKey;
+		if (p_Qty.signum() > 0)
+		{
+			adMessageKey = MSG_Event_OverageGenerated;
+			docTypeQuery.docSubType(DOCSUBTYPE_InventoryOverageDocument);
+		}
+		else if (p_Qty.signum() < 0)
+		{
+			adMessageKey = MSG_Event_ShortageGenerated;
+			docTypeQuery.docSubType(DOCSUBTYPE_InventoryShortageDocument);
+		}
+		else
+		{
+			throw new AdempiereException(MSG_ERROR_QUANTITY_ZERO);
+		}
+
+		final I_M_Inventory inventory = generateInventoryDocument(docTypeDAO.getDocTypeId(docTypeQuery.build()), deliveryPlanningId);
+		createUserNotification(inventory, adMessageKey);
+
+		return MSG_OK;
+	}
+
+	private I_M_Inventory generateInventoryDocument(@NonNull final DocTypeId doctypeId, @NonNull final DeliveryPlanningId deliveryPlanningId)
+	{
+		final DeliveryPlanningReceiptInfo receiptInfo = deliveryPlanningService.getReceiptInfo(deliveryPlanningId);
+		final InOutId inOutId = receiptInfo.getReceiptId();
+		Check.assumeNotNull(inOutId, "InOutId shall be set, because of isReceived() check in preconditions");
+
+		final I_M_InOut inOutRecord = load(inOutId, I_M_InOut.class);
+		final List<I_M_InOutLine> inOutLines = inOutDAO.retrieveLines(inOutRecord);
```

Comment:
instead of loading the lines (=> see comment below) you can invoke IInOutDAO.retrieveLinesForInOutId and just get the IDs - not the whole uneeded record